File: Android_kotlin_app_MVI/app/src/main/java/com/mindsilence/game/navigation/AppUiState.kt
package com.mindsilence.game.navigation // Navigation flags for AppViewModel / AppRoute.

import androidx.compose.runtime.Immutable // Stable snapshot for Compose equality.

/** Which top-level screen to show: training and/or highscores over the menu. */
@Immutable // Compose can skip work when both flags are unchanged.
data class AppUiState( // Two flags; not a back stack.
    val inTraining: Boolean = false, // True after OpenTraining; stays true under highscores.
    val showHighScores: Boolean = false, // True after OpenHighScores; takes precedence in AppRoute.
) // End AppUiState.

File: Android_maui_app/Presentation/Navigation/AppUiState.cs
namespace MindSilence.Presentation.Navigation;

/// <summary>
/// App-wide navigation flags. Visible page: highscores → training → menu.
/// <c>InTraining</c> stays true while highscore is shown on top of training.
/// </summary>
public readonly record struct AppUiState(bool InTraining, bool ShowHighScores)
{
	public bool ShowMenu => !ShowHighScores && !InTraining;

	public bool ShowTraining => InTraining && !ShowHighScores;
}
